src/features/import/hooks/useUIDataColumns.tsx

```diff
@@ -0,0 +1,433 @@
+
+    const uniqueValues = Array.from(new Set(rowsWithValues));
+
+    let columnValuesMessage = '';
+
+    if (numberOfEmptyRows > 0 && uniqueValues.length == 0) {
+      columnValuesMessage = messages.configuration.mapping.messages.onlyEmpty({
+        numEmpty: numberOfEmptyRows,
+      });
+    } else if (numberOfEmptyRows > 0 && uniqueValues.length == 1) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.oneValueAndEmpty({
+          firstValue: uniqueValues[0],
+          numEmpty: numberOfEmptyRows,
+        });
+    } else if (numberOfEmptyRows > 0 && uniqueValues.length == 2) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.twoValuesAndEmpty({
+          firstValue: uniqueValues[0],
+          numEmpty: numberOfEmptyRows,
+          secondValue: uniqueValues[1],
+        });
+    } else if (numberOfEmptyRows > 0 && uniqueValues.length == 3) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.threeValuesAndEmpty({
+          firstValue: uniqueValues[0],
+          numEmpty: numberOfEmptyRows,
+          secondValue: uniqueValues[1],
+          thirdValue: uniqueValues[2],
+        });
+    } else if (numberOfEmptyRows > 0 && uniqueValues.length > 3) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.manyValuesAndEmpty({
+          firstValue: uniqueValues[0],
+          numEmpty: numberOfEmptyRows,
+          numMoreValues: uniqueValues.length - 3,
+          secondValue: uniqueValues[1],
+          thirdValue: uniqueValues[2],
+        });
+    } else if (numberOfEmptyRows == 0 && uniqueValues.length == 1) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.oneValueNoEmpty({
+          firstValue: uniqueValues[0],
+        });
+    } else if (numberOfEmptyRows == 0 && uniqueValues.length == 2) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.twoValuesNoEmpty({
+          firstValue: uniqueValues[0],
+          secondValue: uniqueValues[1],
+        });
+    } else if (numberOfEmptyRows == 0 && uniqueValues.length == 3) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.threeValuesNoEmpty({
+          firstValue: uniqueValues[0],
+          secondValue: uniqueValues[1],
+          thirdValue: uniqueValues[2],
+        });
+    } else if (numberOfEmptyRows == 0 && uniqueValues.length > 3) {
+      columnValuesMessage =
+        messages.configuration.mapping.messages.manyValuesNoEmpty({
+          firstValue: uniqueValues[0],
+          numMoreValues: uniqueValues.length - 3,
+          secondValue: uniqueValues[1],
+          thirdValue: uniqueValues[2],
+        });
+    }
+
+    const valueInFirstRow = cellValues[0];
+    const title =
+      firstRowIsHeaders && valueInFirstRow != null
+        ? valueInFirstRow.toString()
+        : messages.configuration.mapping.defaultColumnHeader({
+            columnIndex: index,
+          });
```

Comment:
Nice! I like this in the hook instead of the store. It means the store state is raw and contains no duplication.